### PR TITLE
add SEMAPHORE_AGENT_TAG_NAME parameter

### PR DESCRIPTION
**Pull request by cchristous**
This adds the `SEMAPHORE_AGENT_TAG_NAME` parameter which allows a user to pass in an argument which gets assigned to the `Name` tag for all resources.

### CODE REVIEW

File: bin/aws-semaphore-agent.js [hunk 1]
```diff
@@ -20,6 +20,10 @@ new AwsSemaphoreAgentStack(app, 'AwsSemaphoreAgentStack', {
   },
 });
 
+if (!argumentStore.isEmpty("SEMAPHORE_AGENT_TAG_NAME")) {
+  Tags.of(awsSemaphoreAgentDedicatedHostStack).add('Name', argumentStore.get("SEMAPHORE_AGENT_TAG_NAME"));
```
Comment (lucaspin) [inline bin/aws-semaphore-agent.js:24]: What is `awsSemaphoreAgentDedicatedHostStack`?

Comment (cchristous) [reply bin/aws-semaphore-agent.js:24]: a copy/paste error. will fix

File: bin/aws-semaphore-agent.js [hunk 2]
```diff
@@ -20,6 +20,10 @@ new AwsSemaphoreAgentStack(app, 'AwsSemaphoreAgentStack', {
   },
 });
 
+if (!argumentStore.isEmpty("SEMAPHORE_AGENT_TAG_NAME")) {
+  Tags.of(awsSemaphoreAgentStack).add('Name', argumentStore.get("SEMAPHORE_AGENT_TAG_NAME"));
```
Comment (lucaspin) [inline bin/aws-semaphore-agent.js:24]: I think you'd need to require `Tags` for this

**Comment by lucaspin** [comment]
What if we go with a more general approach, allowing key-value pairs of tags to be specified, like `SEMAPHORE_AGENT_TAGS='Name:Something,Category:SomethingElse'`?

File: bin/aws-semaphore-agent.js (hunk 2 above)
Comment (cchristous) [reply bin/aws-semaphore-agent.js:24]: yeah, you are right

**Comment by lucaspin** [comment]
@cchristous code looks good. Can you add some unit tests to cover this new `SEMAPHORE_AGENT_TAGS` parameter?

**Comment by cchristous** [comment]
> @cchristous code looks good. Can you add some unit tests to cover this new `SEMAPHORE_AGENT_TAGS` parameter?

added

**Review by lucaspin — APPROVED** [review]

**Pull request merged**